win10/RemoteBlinky/RemoteBlinky/ConnectionPage.xaml.cs

```diff
@@ -0,0 +1,338 @@
+                    } ) );
+                } );
+            }
+        }
+
+        /****************************************************************
+         *                       UI Callbacks                           *
+         ****************************************************************/
+
+        /// <summary>
+        /// This function is called if the selection is changed on the Connection combo box
+        /// </summary>
+        /// <param name="sender">The object invoking the event</param>
+        /// <param name="e">Arguments relating to the event</param>
+        private void ConnectionComboBox_SelectionChanged( object sender, SelectionChangedEventArgs e )
+        {
+            RefreshDeviceList();
+        }
+
+        /// <summary>
+        /// Called if the Refresh button is pressed
+        /// </summary>
+        /// <param name="sender">The object invoking the event</param>
+        /// <param name="e">Arguments relating to the event</param>
+        private void RefreshButton_Click( object sender, RoutedEventArgs e )
+        {
+            RefreshDeviceList();
+        }
+
+        /// <summary>
+        /// Called if the Cancel button is pressed
+        /// </summary>
+        /// <param name="sender">The object invoking the event</param>
+        /// <param name="e">Arguments relating to the event</param>
+        private void CancelButton_Click( object sender, RoutedEventArgs e )
+        {
+            OnConnectionCancelled();
+        }
+
+        /// <summary>
+        /// Called if the Connect button is pressed
+        /// </summary>
+        /// <param name="sender">The object invoking the event</param>
+        /// <param name="e">Arguments relating to the event</param>
+        private void ConnectButton_Click( object sender, RoutedEventArgs e )
+        {
+            //disable the buttons and set a timer in case the connection times out
+            SetUiEnabled( false );
+
+            DeviceInformation device = null;
+            if( ConnectionList.SelectedItem != null )
+            {
+                var selectedConnection = ConnectionList.SelectedItem as Connection;
+                device = selectedConnection.Source as DeviceInformation;
+            }
+            else if( ( ConnectionMethodComboBox.SelectedItem as string ) != "Network" )
+            {
+                //if they haven't selected an item, but have chosen "usb" or "bluetooth", we can't proceed
+                ConnectMessage.Text = "You must select an item to proceed.";
+                SetUiEnabled( true );
+                return;
+            }
+
+            //connection properties dictionary, used only for telemetry data
+            var properties = new Dictionary<string, string>();
+
+            //use the selected device to create our communication object
+            switch( ConnectionMethodComboBox.SelectedItem as string )
+            {
+                default:
+                case "Bluetooth":
```

Comment:
It would be better to use a constant rather than repeating hard coded strings for the connection types.  ie; const string BLUETOOTH = "Bluetooth";   ... case BLUETOOTH: